src/BasePaintSubscription.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: MIT
+
+//  ____                  ____       _       _
+// | __ )  __ _ ___  ___ |  _ \ __ _(_)_ __ | |_
+// |  _ \ / _` / __|/ _ \| |_) / _` | | '_ \| __|
+// | |_) | (_| \__ \  __/|  __/ (_| | | | | | |_
+// |____/ \__,_|___/\___||_|   \__,_|_|_| |_|\__|
+//  ____        _                  _       _   _
+// / ___| _   _| |__  ___  ___ _ __(_)_ __ | |_(_) ___  _ __
+// \___ \| | | | '_ \/ __|/ __| '__| | '_ \| __| |/ _ \| '_ \
+//  ___) | |_| | |_) \__ \ (__| |  | | |_) | |_| | (_) | | | |
+// |____/ \__,_|_.__/|___/\___|_|  |_| .__/ \__|_|\___/|_| |_|
+//                                   |_|
+
+pragma solidity ^0.8.17;
+
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import {IERC1155} from "openzeppelin-contracts/contracts/token/ERC1155/IERC1155.sol";
+import {ERC1155} from "openzeppelin-contracts/contracts/token/ERC1155/ERC1155.sol";
+
+interface IBasePaint is IERC1155 {
+    function openEditionPrice() external view returns (uint256);
+    function mint(uint256 day, uint256 count) external payable;
+    function today() external view returns (uint256);
+}
+
+contract BasePaintSubscription is Ownable, ERC1155 {
+    IBasePaint public basepaint;
+
+    event Subscribed(address indexed subscriber, address indexed mintTo, uint256 length);
+
+    error NotEnoughMinted();
+    error WrongEthAmount();
+    error InsufficientBalance();
+
+    constructor(address _basepaint, address _owner)
+        Ownable(_owner)
+        ERC1155("https://basepaint.xyz/api/subscription/{id}")
+    {
+        basepaint = IBasePaint(_basepaint);
+    }
+
+    function subscribe(uint8 _mintPerDay, address _mintToAddress, uint256 _length) external payable {
```

Comment:
this is very inflexible as there's no way to sub for < 1/day, or choose the start date.

Instead, this should be something like

```solidity
function subscribe(uint256[] calldata _days, address _mintToAddress) external {
```

and let the frontend choose which days to mint